app/views/compute_resources_vms/form/_volume.html.erb

```diff
@@ -1,6 +1,20 @@
 <div class="fields">
-
-  <%= selectable_f f, :sr_uuid, compute_resource.storage_pools.map { |item| [item[:display_name], item[:uuid]] }, { :selected => attribute_map[:volume_selected] }, :class => "span2", :label => _("Storage Repository") %>
+  <%= selectable_f f, :sr_uuid,
```

Comment:
If you want, you could create a new helper method `selectable_f_with_cache_invalidation` that calls `selectable_f` and adds the refresh stuff. That way you don't have to repeat yourself so often.